sklearn/linear_model/tests/test_logistic.py

```diff
@@ -2057,3 +2057,29 @@ def test_liblinear_not_stuck():
     with warnings.catch_warnings():
         warnings.simplefilter("error", ConvergenceWarning)
         clf.fit(X_prep, y)
+
+
+@pytest.mark.parametrize("solver", SOLVERS)
+def test_null_max_iter(solver):
+    # Make sure we can inspect the state of LogisticRegression right after
+    # initialization (before the first weight update).
+    X, y = load_iris(return_X_y=True)
+    y = y == 2
+    with ignore_warnings(category=ConvergenceWarning):
+        clf = LogisticRegression(solver=solver, max_iter=0).fit(X, y)
+    if solver not in ["saga", "sag"]:
+        # XXX: sag and saga have n_iter_ = [1]...
+        assert clf.n_iter_ == 0
+
+    if solver != "lbfgs":
+        # XXX: lbfgs has already started to update the coefficients...
+        assert_allclose(clf.coef_, np.zeros_like(clf.coef_))
```

Comment:
I‘m not sure whether we need this. We might decide to initialize coefficients different from zero.

Reply:
Do you really anticipate a non-zero init for the coefficients of linear models? I have personally never seen that and I don't think we should depart from this scheme unless we have a really good reason to and at this point we can just change this test.

Reply:
AFAIK, coefficients in GLMs are often initialized non-zero in IRLS, e.g. in R.
But my point is more to make the test only test what it needs to test.
If you really want to keept it, fine for me.

Reply:
Out of curiosity, what init would that be. The marginal data mean for binomial and the per-class means for multinomial?

I think we can keep the test this way to check that `max_iter=0` does not mutates the coef values beyond their init and make sure that all solvers behave consistently in that respect (the `XXX` comments highlight that is not yet the case).

Reply:
If you really want I can update to check that all solvers yield consistent `coef_` / `intercept_`, `predict_proba` / `decision_function` when fitting on the same data with `max_iter=0`. But since to me 0 coef sounds like the expected init at the moment and 0.5/0.5 binary class probabilities, I think it's even better to write that in the test directly.

We can always change our minds later and update the test accordingly. Having to change the test will make think that we need to document such a behavioral change in the changelog if decide to do it.

Reply:
For the curiosity: With classical IRLS, one can start with linear predictor = raw predictions = inverse link of observed values

Reply:
So let’s keep it.